Sources/bonfire/flame/Lexer.hx

```diff
@@ -0,0 +1,189 @@
+                        tokens.push(new Token(TokenType.LESS, str, index));
+                    continue;
+                case "=":
+                    if (siter.eatIf("="))
+                        tokens.push(new Token(TokenType.EQL_EQL, str, index));
+                    else
+                        tokens.push(new Token(TokenType.SET, str, index));
+                    continue;
+                
+                case "!":
+                    tokens.push(new Token(TokenType.BANG, str, index));
+                    continue;
+
+                case "\"":
+                    var value = "";
+                    for (char in siter) {
+                        if (char == "\"")
+                            break;
+                        if (char == "\\") {
+                            var nextChar = siter.next();
+                            switch (nextChar) {
+                                case "t":
+                                    value += "\t";
+                                case "n":
+                                    value += "\n";
+                                default:
+                                    value += nextChar;
+                            }
+                        } else {
+                            value += char;
+                        }
+                    }
+                    tokens.push(new Token(TokenType.STRING, new StringLiteral(value), str, index));
+                    continue;
+                case "\'":
+                    var value = "";
+                    for (char in siter) {
+                        if (char == "\'")
+                            break;
+                        if (char == "\\") {
+                            var nextChar = siter.next();
+                            switch (nextChar) {
+                                case "t":
+                                    value += "\t";
+                                case "n":
+                                    value += "\n";
+                                default:
+                                    value += nextChar;
+                            }
+                        } else {
+                            value += char;
+                        }
+                    }
+                    tokens.push(new Token(TokenType.STRING, new StringLiteral(value), str, index));
+                    continue;
+                
+                case ",":
+                    tokens.push(new Token(TokenType.COMMA, str, index));
+                    continue;
+            }
+
+            if (numberRegex.match(siter.string.substr(siter.index - 1))) {
+                tokens.push(new Token(TokenType.NUMBER, new NumberLiteral(Std.parseFloat(numberRegex.matched(0))), str, index));
+                siter.index += numberRegex.matchedPos().len - 1;
+                continue;
+            }
+            
+            if (symbolRegex.match(siter.string.substr(siter.index - 1))) {
+                switch (symbolRegex.matched(0)) {
```

Comment:
Similarly, you could do `tokens.push(switch ...)` instead of repeating the call in each branch.